### PR TITLE
CFE-2898: validdata/validjson examples

### PR DESCRIPTION
**Pull request by karlhto**
Also fixed wrong return type for `validjson`

### CODE REVIEW

**Comment by karlhto** [comment]
@cf-bottom Jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4638/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4638/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4638/

File: examples/validdata.cf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  vars:
+    "json_string" string => '{"test": [1, 2, 3]}';
+
+  classes:
+    "valid" expression => validdata("$(json_string)", "JSON");
+
+  reports:
+    valid::
+      "Valid json string";
+    !valid::
```
Comment (olehermanse) [inline examples/validdata.cf]: As indicated in the other PR, don't use classes and `!` like this. Either use separate bundles to force ordering, or use other means to make decisions, like `if => validjson`. This has the advantage of function / promise skipping during early evaluation passes.

https://docs.cfengine.com/docs/master/reference-language-concepts-classes.html#negative-knowledge

https://docs.cfengine.com/docs/3.13/reference-functions.html#function-skipping

**Comment by karlhto** [comment]
@cf-bottom Jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4641/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4641/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4641/

**Comment by karlhto** [comment]
Only change is category of `validdata` and `validjson`, so no rerun of tests needed : )

**Pull request merged**